src/UrdFactory.sol

```diff
@@ -22,8 +23,8 @@ contract UrdFactory {
         bytes32 initialRoot,
         bytes32 initialIpfsHash,
         bytes32 salt
-    ) public returns (address urd) {
-        urd = address(
+    ) public returns (IUniversalRewardsDistributor urd) {
+        urd = IUniversalRewardsDistributor(
```

Comment:
I think typing it is unnecessary as `UniversalRewardsDistributor is IUniversalRewardsDistributor`